applications/RANSApplication/python_scripts/block_builder_and_solvers.py

```diff
@@ -0,0 +1,29 @@
+import KratosMultiphysics as Kratos
+import KratosMultiphysics.FluidDynamicsApplication as KratosCFD
+
+from KratosMultiphysics import IsDistributedRun
+from KratosMultiphysics.kratos_utilities import CheckIfApplicationsAvailable
+
+if (IsDistributedRun()
+        and CheckIfApplicationsAvailable("TrilinosApplication")):
+    import KratosMultiphysics.TrilinosApplication as KratosTrilinos
+
+
+def TrilinosPeriodicBlockBuilderAndSolver(linear_solver, communicator):
+    return KratosTrilinos.TrilinosBlockBuilderAndSolverPeriodic(
+        communicator, 30, linear_solver,
+        KratosCFD.PATCH_INDEX)
+
+
+def TrilinosBlockBuilderAndSolver(linear_solver, communicator):
+    return KratosTrilinos.TrilinosBlockBuilderAndSolver(
+        communicator, 30, linear_solver)
+
+
+def PeriodicBlockBuilderAndSolver(linear_solver, _):
+    return KratosCFD.ResidualBasedBlockBuilderAndSolverPeriodic(
+        linear_solver, KratosCFD.PATCH_INDEX)
+
+
+def BlockBuilderAndSolver(linear_solver, _):
+    return Kratos.ResidualBasedBlockBuilderAndSolver(linear_solver)
```

Comment:
Shouldn't these be implemented in the `_create_builder_and_solver` of the solvers?

Reply:
The idea is, I wanted to have common interface for OpenMP and MPI `BuilderAndSolver` ctors so, formulations can create them without considering whether the run is in OpenMP or MPI. (details of formulation can be found in #7304.)